### PR TITLE
Remove component and fix urls

### PR DESCRIPTION
**Pull request by RongxinZhang**
**Issue:**
- Current backbone is working with components which is no longer supported
- It is also using an old version js-sdk and it can't be updated because it's using components
- Endpoints pluralization that has not been added
- URL is currently  `lists/<vendor>` should be `/<vendor>/lists`.

**Solution:**
- [x] Remove components and use npm
- [x] Make Retsly js Sdk work with browserify
- [x] Fix urls
- [x] Make tests run (Optional) 
- [x] Add package to npm

### CODE REVIEW

**Comment by RongxinZhang** [comment]
@Jkhwan @lvivier please review.

File: Makefile [hunk 1]
```diff
@@ -1,18 +1,12 @@
...
-
-test: build
-	@mocha-phantomjs test/test.html
+test:
+	@npm run build
+	@npm test
 
 clean:
-	@rm -fr build components
+	@rm -fr node_modulesbuild components
```
Comment (lvivier) [inline Makefile]: typo (`node_modules build`)

File: README.md [hunk 2]
```diff
@@ -6,23 +6,106 @@ Use [Retsly](https://rets.ly/) with Backbone.
 
 ## Install
 
-Install with [component](https://github.com/component/component):
-
-    $ component install retsly/retsly-js-backbone
+Javascript
```
Comment (lvivier) [inline README.md]: Should this say something like "Install with npm:"?

Comment (RongxinZhang) [inline README.md]: @lvivier yes, will add this

File: test/test.html [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
-      require('mocha');
-      mocha.setup('qunit');
+      mocha.ui('bdd');
+      mocha.reporter('html');
+      var expect = chai.expect;
     </script>
     <script src="./index.js"></script>
     <script>
-    window.mochaPhantomJS ?mochaPhantomJS.run() : mocha.run();
+      console.log(Retsly);
```
Comment (lvivier) [inline test/test.html]: stray console.log?

**Comment by Jkhwan** [comment]
looks good to me :+1:

**Comment by lvivier** [comment]
:+1:

**Comment by RongxinZhang** [comment]
Merging.

**Comment by RongxinZhang** [comment]
This image is just for hosting:
screen shot 2015-11-03 at 2 27 56 pm

**Pull request merged**